tests/interactions/test_base.py

```diff
@@ -116,3 +125,80 @@ async def test_defer_invalid_parent(self, response: disnake.InteractionResponse,
         with pytest.raises(TypeError, match="This interaction must be of type"):
             await response.defer()
         adapter.create_interaction_response.assert_not_called()
+
+
+class TestInteractionDataResolved:
+    # TODO: use proper mock models once we have state/guild mocks
+    @pytest.fixture()
+    def state(self):
+        s = mock.Mock(spec_set=ConnectionState)
+        s._get_guild.return_value = None
+        return s
+
+    def test_init_member(self, state):
+        member_payload: MemberPayload = {
+            "roles": [],
+            "joined_at": "2022-09-02T22:00:55.069000+00:00",
+            "deaf": False,
+            "mute": False,
+        }
+
+        user_payload: UserPayload = {
+            "id": "1234",
+            "discriminator": "1111",
+            "username": "h",
+            "avatar": None,
+        }
+
+        # user only, should deserialize user object
+        resolved = disnake.InteractionDataResolved(
+            data={"users": {"1234": user_payload}},
+            state=state,
+            guild_id=1234,
+        )
+        assert len(resolved.members) == 0
+        assert len(resolved.users) == 1
+
+        # member only, shouldn't deserialize anything
+        resolved = disnake.InteractionDataResolved(
+            data={"members": {"1234": member_payload}},
+            state=state,
+            guild_id=1234,
+        )
+        assert len(resolved.members) == 0
+        assert len(resolved.users) == 0
+
+        # user + member, should deserialize member object only
+        resolved = disnake.InteractionDataResolved(
+            data={"users": {"1234": user_payload}, "members": {"1234": member_payload}},
+            state=state,
+            guild_id=1234,
+        )
+        assert len(resolved.members) == 1
+        assert len(resolved.users) == 0
+
+    @pytest.mark.parametrize("channel_type", [t.value for t in disnake.ChannelType])
+    def test_channel(self, state, channel_type):
+        channel_data: ResolvedPartialChannelPayload = {
+            "id": "42",
+            "type": channel_type,
+            "permissions": "7",
+            "name": "a-channel",
+        }
+        if channel_type in (10, 11, 12):  # thread
+            channel_data["parent_id"] = "123123"
+            channel_data["thread_metadata"] = {
+                "archived": False,
+                "auto_archive_duration": 60,
+                "archive_timestamp": "2022-09-02T22:00:55.069000+00:00",
+                "locked": False,
+            }
+
+        resolved = disnake.InteractionDataResolved(
+            data={"channels": {"42": channel_data}}, state=state, guild_id=1234
+        )
+        assert len(resolved.channels) == 1
+
+        channel = next(iter(resolved.channels.values()))
+        # should be partial iff dm/group
```

Comment:
```suggestion
        # should be partial if dm/group
```

Reply:
not a typo[^1], but updated in https://github.com/DisnakeDev/disnake/pull/814/commits/eb59a4185290afc4e149e455d93ee3e76d74313f

[^1]: https://en.wikipedia.org/wiki/If_and_only_if